### ISSUE TITLE
F5 Pool nodes are not configured with the specified route domain by the k8s-bigip-ctrl

### ISSUE THREAD

**Issue by diaconud007** [F5Networks/k8s-bigip-ctlr#1315]
Hello F5Networks Team,

I have a k8s cluster were I configured a k8s-bigip-ctrl-er. I have the following configuration:

ConfigMap:

``` kind: ConfigMap
apiVersion: v1
metadata:
  name: istio-ingress-vsphere-vs-80-cm-crmk8s-reloaded
  namespace: istio-system
  labels:
    f5type: virtual-server
data:
  # See the f5-schema table for schema-controller compatibility
  # https://clouddocs.f5.com/containers/latest/releases_and_versioning.html#f5-schema
  schema: "f5schemadb://bigip-virtual-server_v0.1.7.json"
  data: |
    {
      "virtualServer": {
        "backend": {
          "servicePort": 80,
          "serviceName": "istio-ingressgateway",
          "healthMonitors": [
          {
            "protocol": "tcp",
            "interval": 5,
            "timeout": 16
          }
         ]
        },
        "frontend": {
          "virtualAddress": {
            "port": 80,
            "bindAddr": "$IP%52"
          },
          "partition": "rancherpoc-usercluster3",
          "balance": "round-robin",
          "mode": "http"
        }
      }
    }
```
where 52 is the ID of the routing domain.

Controller:

```
apiVersion: apps/v1
kind: Deployment
metadata:
  name: k8s-bigip-ctlr-deployment
  namespace: kube-system
spec:
  # DO NOT INCREASE REPLICA COUNT
  replicas: 1
  selector:
    matchLabels:
      app: k8s-bigip-ctlr
  template:
    metadata:
      name: k8s-bigip-ctlr
      labels:
        app: k8s-bigip-ctlr
    spec:
      # Name of the Service Account bound to a Cluster Role with the required
      # permissions
      serviceAccountName: bigip-ctlr
      containers:
        - name: k8s-bigip-ctlr
          image: "f5networks/k8s-bigip-ctlr:1.14.0"
          env:
            - name: BIGIP_USERNAME
              valueFrom:
                secretKeyRef:
                  # Replace with the name of the Secret containing your login
                  # credentials
                  name: bigip-login
                  key: username
            - name: BIGIP_PASSWORD
              valueFrom:
                secretKeyRef:
                  # Replace with the name of the Secret containing your login
                  # credentials
                  name: bigip-login
                  key: password
          command: ["/app/bin/k8s-bigip-ctlr"]
          args: [
            # See the k8s-bigip-ctlr documentation for information about
            # all config options
            # https://clouddocs.f5.com/products/connectors/k8s-bigip-ctlr/latest
            "--bigip-username=$(BIGIP_USERNAME)",
            "--bigip-password=$(BIGIP_PASSWORD)",
            "--bigip-url=$(BIGIP_URL)",
            "--bigip-partition=rancherpoc-usercluster3",
            "--pool-member-type=nodeport",
            "--insecure=true",
            "--default-ingress-ip=$IP%52",
            "--log-level=debug"
            ]
      imagePullSecrets:
        # Secret that gives access to a private docker registry
        - name: f5-docker-images
        # Secret containing the BIG-IP system login credentials
        - name: bigip-login
```

With this configuration, the virtual server is correctly  configured -> with route domain suffix (in my case with %52), but the nodes aren't configured as I expect  -> they are configured with the partition default routing domain.

Is this a normal behavior or is something that is missing from my configuration?

Thank you,
DD

**Comment by mdditt2000** [comment]
I am surprised that route domain is not working with mode CCCL. Looking into this.

Known issue with CIS using AS3 which will be resolved in CIS 2.1. Workaround is to use AS3 override https://github.com/F5Networks/k8s-bigip-ctlr/issues/1287

**Comment by veeramalla-f5** [comment]
I could reproduce this issue. Assigning it to myself for further analysis.

**Comment by mdditt2000** [comment]
Created internal Jira CONTCNTR-1842 for PM tracking

**Comment by mdditt2000** [comment]
CIS with agent cccl Route domain for pool members are getting updated but with AS3 it's not updated. The issue is reproduced.

**Comment by mdditt2000** [comment]
Jira is now closed - CONTCNTR-1842 - Shipping in CIS 2.1 next month

**Resolution** - CIS with cccl mode with userdefined configmap for VS, route domain is updated properly and for pool member, route domain needs to create the "route domain" on BIG-IP first for that partition and assign it using default route domain. We have an automation test case for this and its getting pass.

Closing issue out as complete
